experimental/devices/ina219/ina219.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package ina219
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"sync"
+
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/mmr"
+	"periph.io/x/periph/conn/physic"
+)
+
+// Config is a Configuration
+type Config struct {
+	Address       int
+	SenseResistor physic.ElectricResistance
+	MaxCurrent    physic.ElectricCurrent
+}
+
+const (
+	// DefaultSenseResistor is 100 mΩ and can be changed using SenseResistor(value)
+	DefaultSenseResistor = 100 * physic.MilliOhm
+	// DefaultI2CAddress is 0x40 and can be changed using Address(address)
+	DefaultI2CAddress = 0x40
+	// DefaultMaxCurrent is 3.2 and can be changed using MaxCurrent(value)
+	DefaultMaxCurrent = 3200 * physic.MilliAmpere
+)
+
+// New opens a handle to an ina219 sensor
+func New(bus i2c.Bus, config Config) (*Dev, error) {
+
+	i2cAddress := DefaultI2CAddress
+	if config.Address != 0 {
+		if config.Address < 0x40 || config.Address > 0x4f {
+			return nil, errAddressOutOfRange
+		}
+		i2cAddress = config.Address
+	}
+
+	senseResistor := DefaultSenseResistor
+	if config.SenseResistor != 0 {
+		if config.SenseResistor < 1 {
+			return nil, errSenseResistorValueInvalid
+		}
+		senseResistor = config.SenseResistor
+	}
+
+	maxCurrent := DefaultMaxCurrent
+	if config.MaxCurrent != 0 {
+		if config.MaxCurrent < 1 {
+			return nil, errMaxCurrentInvalid
+		}
+		maxCurrent = config.MaxCurrent
+	}
+
+	dev := &Dev{
+		m: &mmr.Dev8{
+			Conn:  &i2c.Dev{Bus: bus, Addr: uint16(i2cAddress)},
+			Order: binary.BigEndian,
+		},
+	}
+
+	if err := dev.Calibrate(senseResistor, maxCurrent); err != nil {
+		return nil, err
+	}
+
+	if err := dev.m.WriteUint16(configRegister, 0x1FFF); err != nil {
+		return nil, errWritingToConfigRegister
+	}
+
+	return dev, nil
+}
+
+// Dev is a handle to the ina219 sensor.
+type Dev struct {
+	m *mmr.Dev8
```

Comment:
This doesn't need to be a pointer, since mmr.Dev8 is a simple struct, so
`m  mmr.Dev8`

This reduces heap usage, increase performance.